data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v1/events/EventSubscriptionManagerTest.java

```diff
@@ -392,4 +340,15 @@ private void triggerContributionEvent() {
         contributionAndProof, InternalValidationResult.ACCEPT, false);
     asyncRunner.executeQueuedActions();
   }
+
+  private <T, E extends Event<T>> void checkEvent(String eventType, E event)
+      throws JsonProcessingException {
+    final String eventString = outputStream.getString();
+    assertThat(eventString).contains(String.format("event: %s\n", eventType));
+
+    String expected =
+        eventString.substring(eventString.indexOf("{"), eventString.lastIndexOf("}") + 1);
+    String result = JsonUtil.serialize(event.getData(), event.getJsonTypeDefinition());
+    assertThat(expected).isEqualTo(result);
```

Comment:
putting aside the nit of using `final`...

We've got a tiny bit of a design problem with the previous test here that sneaks into this test for the same reason...

Maybe we can do a part 2 of this where the json is compared to a file if the data is static... then new fields break the test...

a practical example would be adding in `execution_optimistic`, and this test will not fail, even though the data changed...